### PR TITLE
fix(): stop transforming on `discardActiveObject`

### LINKED ISSUE #7953: `discardActiveObject` is not prevent moving of object
When i start move object with mouse and call `discardActiveObject`, the active object are reset, but i still move object.
I can't stop moving object programmatically.

## Version
5.0.0

## Test Case
Browser: https://jsfiddle.net/aLuvj3hp/

## Steps To Reproduce

1. Start move red object
2. Wait 3 second, to selection will reset

## Expected Behavior

- Active object are reseted
- Moving of object are stopped

## Actual Behavior

- Active object are reseted
- User still can move object

### PR DESCRIPTION
**Pull request by ShaMan123**
Motivation 
closes #7953

### CODE REVIEW

File: src/canvas.class.js [hunk 1]
```diff
@@ -1243,6 +1243,10 @@
         if (obj.onDeselect({ e: e, object: object })) {
           return false;
         }
+        if (this._currentTransform && this._currentTransform.target === obj) {
```
Comment (ShaMan123) [inline src/canvas.class.js:1246]: pretty sure `this._currentTransform.target === obj` is useless and harmless

File: src/canvas.class.js [hunk 2]
```diff
@@ -1243,6 +1243,11 @@
...
+          this._finalizeCurrentTransform(e);
+          this._currentTransform = null;
```
Comment (ShaMan123) [inline src/canvas.class.js:1248]: should we call `this._resetTransformEventData();`?

**Comment by asturur** [comment]
I m ok with this PR but i would like there was a canvas method that says `(cancel/stop/end)CurrentTransform` and does what it needs to do to to cancel the current transform, and we use it on discardActiveObject

**Comment by asturur** [comment]
i grouped the 2 calls in a method. i m ok merginging this.
The only doubt i have left and we can solve whenever is if that the discard of active objects belong to _discardActiveObject or discardActiveObject.

for example on object removed calls _discardActiveObject, and it could call _discardActiveObject and endCurrentTransform, instead of relying on the inner part, focused on discardind, now also fires an object modified event ( eventually ).

Is not a big deal, can probably be changed later.
Too much automatic behaviours usually brought to issues in the past

**Review by asturur — APPROVED** [review]

**Comment by ShaMan123** [comment]
> for example on object removed calls _discardActiveObject, and it could call _discardActiveObject and endCurrentTransform, instead of relying on the inner part, focused on discardind, now also fires an object modified event ( eventually ).

Sounds a good point to me.

I am not sure but I think I walked into a bug with this PR somewhere...
Well, I'll try reproduce and PR a fix is so.

**Comment by asturur** [comment]
is ok to merge this part as it is, is reasonably correct and help us moving forward

**Comment by asturur** [comment]
is small, is focused, has a test, has docs, can be merged imho.

File: src/canvas.class.js (hunk 2 above)
Comment (ShaMan123) [reply src/canvas.class.js]: @asturur this is not outdated
Thoughts?

Comment (ShaMan123) [reply src/canvas.class.js]: It's really `Canvas._target` , I think it should be reset but I'm not sure

Comment (ShaMan123) [reply src/canvas.class.js]: I have encountered the case that demands to nullify `_target`
When an object is being moved by the user and discarded meanwhile.
Currently the object will continue moving and will be inactive only on mouseup.

Comment (ShaMan123) [reply src/canvas.class.js]: Note for future self

setting `canvas._target` on `mousedown:before` hijacks the event and fires it on the hijacker

Comment (ShaMan123) [reply src/canvas.class.js]: #8335

**Pull request merged**